native/src/main/java/io/ballerina/stdlib/data/xml/Native.java

```diff
@@ -19,25 +26,81 @@
  */
 public class Native {
 
+    private static final String METHOD_NAME_NEXT = "next";
+    private static final String METHOD_NAME_CLOSE = "close";
+
     public static Object fromXmlWithType(BXml xml, BMap<BString, Object> map, BTypedesc typed) {
         return null;
     }
 
-    public static Object fromXmlStringWithType(Object xml, BMap<BString, Object> map, BTypedesc typed) {
+    public static Object fromXmlStringWithType(Environment env, Object xml, BMap<BString, Object> map,
+                                               BTypedesc typed) {
         try {
             if (xml instanceof BString) {
                 return XmlParser.parse(new StringReader(((BString) xml).getValue()), typed.getDescribingType());
             } else if (xml instanceof BArray) {
                 byte[] bytes = ((BArray) xml).getBytes();
-                return XmlParser.parse(
-                        new InputStreamReader(new ByteArrayInputStream(bytes)), typed.getDescribingType());
+                return XmlParser.parse(new InputStreamReader(new ByteArrayInputStream(bytes)),
+                                       typed.getDescribingType());
             } else if (xml instanceof BStream) {
-                return null;
+                final BObject iteratorObj = ((BStream) xml).getIteratorObj();
+                final Future future = env.markAsync();
+                ResultConsumer<Object> resultConsumer = new ResultConsumer<>(future);
+                try (var byteBlockSteam = new BallerinaByteBlockInputStream(env, iteratorObj,
+                                                                            resolveNextMethod(iteratorObj),
+                                                                            resolveCloseMethod(iteratorObj),
+                                                                            resultConsumer)) {
+
+                    Object result = XmlParser.parse(new InputStreamReader(byteBlockSteam), typed.getDescribingType());
+                    future.complete(result);
+                } catch (Exception e) {
+                    return DataUtils.getError("Error occurred while reading the stream: " + e.getMessage());
+                }
+                return DataUtils.getError("Error occurred while reading the stream");
```

Comment:
Here we always returning the error. Right?

Reply:
Good catch it was a bug.